### PR TITLE
Feature/release2 wa1

### PR DESCRIPTION
**Pull request by udaydenduluri33**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-3624
https://tools.hmcts.net/jira/browse/EUI-3566
https://tools.hmcts.net/jira/browse/EUI-3681
https://tools.hmcts.net/jira/browse/EUI-3999

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: api/user/index.spec.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import 'mocha'
 import * as sinon from 'sinon'
 import * as sinonChai from 'sinon-chai'
 import { mockReq, mockRes } from 'sinon-express-mock'
-import { getUserDetails } from './index'
+import { getUserDetails, getUserRoleAssignments } from './index'
```
Comment (adnanakgun) [inline api/user/index.spec.ts:8]: Looks like we are introducing semi-colons at the end of lines on the other files. Why not here?

File: api/user/index.ts [hunk 2]
```diff
@@ -1,7 +1,11 @@
 import { getUserSessionTimeout, RoleGroupSessionTimeout } from '@hmcts/rpx-xui-node-lib'
+import { UserInfo } from 'auth/interfaces/UserInfo'
```
Comment (adnanakgun) [inline api/user/index.ts:2]: Again, semi-colons.

File: src/app/components/hmcts-global-header/hmcts-global-header.component.spec.ts [hunk 3]
```diff
@@ -49,14 +49,14 @@ describe('HmctsGlobalHeaderComponent', () => {
     expect(component).toBeTruthy();
   });
 
-  it('should show the Case search button as inactive when the currentUrl does not match', () => {
+  xit('should show the Case search button as inactive when the currentUrl does not match', () => {
```
Comment (adnanakgun) [inline src/app/components/hmcts-global-header/hmcts-global-header.component.spec.ts:52]: Can we either delete the test or fix the code please? Just don't `xit`. xit is for local use only.

File: src/app/components/hmcts-global-header/hmcts-global-header.component.spec.ts [hunk 4]
```diff
@@ -49,14 +49,14 @@ describe('HmctsGlobalHeaderComponent', () => {
...
     const searchButton = fixture.debugElement.nativeElement.querySelector('.hmcts-search-toggle__button');
     expect(searchButton).toBeDefined();
     expect(searchButton.textContent).toBe('Find case');
     expect(searchButton.getAttribute('aria-current')).not.toEqual('true');
   });
 
-  it('should show the Case search button as active when the currentUrl matches', () => {
+  xit('should show the Case search button as active when the currentUrl matches', () => {
```
Comment (adnanakgun) [inline src/app/components/hmcts-global-header/hmcts-global-header.component.spec.ts:59]: same as above

File: src/app/containers/app-header/app-header.component.ts [hunk 5]
```diff
@@ -161,6 +162,10 @@ export class AppHeaderComponent implements OnInit, OnDestroy {
       const applicationTheme: Theme = this.getApplicationThemeForUser(applicationThemes, userDetails.userInfo.roles);
       this.hideNavigationListener(this.store);
       this.setAppHeaderProperties(applicationTheme);
+      if (applicationTheme.navigationItems.some(x => x.active)) {
+        const currentNav = applicationTheme.navigationItems.find(x => x.active);
+        // this.router.navigate([currentNav.href]);
```
Comment (adnanakgun) [inline src/app/containers/app-header/app-header.component.ts:167]: Is this commenting out on purpose? Please remove if it is not going to be used again.

File: src/app/containers/app/app.component.ts [hunk 6]
```diff
@@ -78,7 +82,7 @@ export class AppComponent implements OnInit, OnDestroy {
   }
 
   public handleCookieBannerFeatureToggle(): void {
-    this.cookieBannerEnabledSubscription = this.featureToggleService.isEnabled('mc-cookie-banner-enabled')
+    this.cookieBannerEnabledSubscription = this.featureService.isEnabled('mc-cookie-banner-enabled')
```
Comment (adnanakgun) [inline src/app/containers/app/app.component.ts:85]: Is this not already in master? Maybe you need to rebase/merge?

Comment (udaydenduluri33) [reply src/app/containers/app/app.component.ts:85]: Its just a rename of featureToggleService to featureService

Comment (adnanakgun) [reply src/app/containers/app/app.component.ts:85]: Ah yes, of course. My bad.

File: src/app/components/hmcts-global-header/hmcts-global-header.component.spec.ts (hunk 4 above)
Comment (udaydenduluri33) [reply src/app/components/hmcts-global-header/hmcts-global-header.component.spec.ts]: Done

File: src/app/components/hmcts-global-header/hmcts-global-header.component.spec.ts (hunk 3 above)
Comment (udaydenduluri33) [reply src/app/components/hmcts-global-header/hmcts-global-header.component.spec.ts]: Done

File: api/user/index.ts (hunk 2 above)
Comment (udaydenduluri33) [reply api/user/index.ts]: Agreed, its done randomly. I have change this now.

File: api/user/index.spec.ts (hunk 1 above)
Comment (udaydenduluri33) [reply api/user/index.spec.ts]: Agreed, its done randomly. I have change this now.

**Comment by adnanakgun** [comment]
@udaydenduluri33 Please also note that src/app/components/routing/application-routing.component.html is empty. Can't comment on the file as there is no code. Am continuing to review, huge PR.

File: src/style/app.scss [hunk 7]
```diff
@@ -388,3 +388,8 @@ button:focus {
     background-color: #ffdd00 !important;
   }
 }
+
+exui-task-list {
+  display: block;
+  min-height: 250px;
+}
```
Comment (adnanakgun) [inline src/style/app.scss:395]: No new line at the end of file. Isn't linting supposed to catch this?

File: src/work-allocation-2/components/available-tasks-filter/available-tasks-filter.component.scss [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+    }
+    .checkbox-items {
+      margin-left: -10px;
+      padding-left: 10px;
+    }
+  }
+}
+.govuk-section-break {
+  height: auto;
+}
```
Comment (adnanakgun) [inline src/work-allocation-2/components/available-tasks-filter/available-tasks-filter.component.scss:17]: Again no new line at the end of file.

File: src/app/containers/app-header/app-header.component.ts (hunk 5 above)
Comment (udaydenduluri33) [reply src/app/containers/app-header/app-header.component.ts]: Very good point. I have removed the redundant logic now. Thanks

File: src/work-allocation-2/components/available-tasks-filter/available-tasks-filter.component.scss (hunk 8 above)
Comment (udaydenduluri33) [reply src/work-allocation-2/components/available-tasks-filter/available-tasks-filter.component.scss:17]: New line is required for .ts file. Not needed for html and scss

File: src/style/app.scss (hunk 7 above)
Comment (udaydenduluri33) [reply src/style/app.scss:395]: New line is required for .ts file. Not needed for html and scss

**Comment by udaydenduluri33** [comment]
> application-routing.component.html

This is a routing component. Nothing to show there

**Comment by adnanakgun** [comment]
> > application-routing.component.html
> 
> This is a routing component. Nothing to show there

Is the template needed then?

**Comment by udaydenduluri33** [comment]
> > > application-routing.component.html
> > 
> > 
> > This is a routing component. Nothing to show there
> 
> Is the template needed then?

Not needed now. If we have a template. But, we could show anything whilst there is delay in routing/network.  I can remove it

File: api/workAllocation2/caseWorkerService.ts [hunk 9]
```diff
@@ -0,0 +1,89 @@
+import { AxiosResponse } from 'axios';
```
[the remark this answers is not in the document]
Comment (udaydenduluri33) [reply api/workAllocation2/caseWorkerService.ts:1]: No unit tests for this atm. I can add it. Do you want me to?

[the remark this answers is not in the document]
Comment (adnanakgun) [reply api/workAllocation2/caseWorkerService.ts:1]: That would be best practice, wouldn't it?

**Comment by adnanakgun** [comment]
> > > > application-routing.component.html
> > > 
> > > 
> > > This is a routing component. Nothing to show there
> > 
> > 
> > Is the template needed then?
> 
> Not needed now. If we have a template. But, we could show anything whilst there is delay in routing/network. I can remove it?

Up to you mate. If you think we can make use of it later, leave it.

**Comment by udaydenduluri33** [comment]
> > > > > application-routing.component.html
> > > > 
> > > > 
> > > > This is a routing component. Nothing to show there
> > > 
> > > 
> > > Is the template needed then?
> > 
> > 
> > Not needed now. If we have a template. But, we could show anything whilst there is delay in routing/network. I can remove it?
> 
> Up to you mate. If you think we can make use of it later, leave it.

I will leave it for now

File: src/work-allocation-2/pipes/caseworker-display-name.pipe.ts [hunk 10]
```diff
@@ -0,0 +1,17 @@
+import { Pipe, PipeTransform } from '@angular/core';
```
Comment (adnanakgun) [inline src/work-allocation-2/pipes/caseworker-display-name.pipe.ts:1]: Are there unit tests for this pipe?

File: src/work-allocation-2/pipes/days-from-today.pipe.ts [hunk 11]
```diff
@@ -0,0 +1,33 @@
+import { Pipe, PipeTransform } from '@angular/core';
```
Comment (adnanakgun) [inline src/work-allocation-2/pipes/days-from-today.pipe.ts:1]: Are there unit tests for this pipe?

File: src/work-allocation-2/pipes/integer.pipe.ts [hunk 12]
```diff
@@ -0,0 +1,27 @@
+import { DecimalPipe } from '@angular/common';
```
Comment (adnanakgun) [inline src/work-allocation-2/pipes/integer.pipe.ts:1]: Are there unit tests for this pipe?

File: src/work-allocation-2/pipes/two-dp.pipe.ts [hunk 13]
```diff
@@ -0,0 +1,27 @@
+import { DecimalPipe } from '@angular/common';
```
Comment (adnanakgun) [inline src/work-allocation-2/pipes/two-dp.pipe.ts:1]: Are there unit tests for this pipe?

File: src/work-allocation-2/pipes/yes-no.pipe.ts [hunk 14]
```diff
@@ -0,0 +1,18 @@
+import { Pipe, PipeTransform } from '@angular/core';
```
Comment (adnanakgun) [inline src/work-allocation-2/pipes/yes-no.pipe.ts:1]: Are there unit tests for this pipe?

**Review by adnanakgun — CHANGES_REQUESTED** [review]
A few minor change requests.

**Review by adnanakgun — DISMISSED** [review]
Thanks for adding the pipe tests @connorpgpmcelroy

**Review by leepavlouhmcts — DISMISSED** [review]

**Review by Mo-Lala84 — APPROVED** [review]
approveddeed

**Review by adnanakgun — APPROVED** [review]

**Review by Mo-Lala84 — APPROVED** [review]
approved

**Review by leepavlouhmcts — APPROVED** [review]

**Review by Mo-Lala84 — APPROVED** [review]
approved

**Review by adnanakgun — APPROVED** [review]

**Review by leepavlouhmcts — APPROVED** [review]

**Pull request merged**